packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,485 @@
+ * const result = await handler.run(channel, author);
+ * ```
+ */
+export class MessagePrompter {
+	/**
+	 * The strategy used in [[MessagePrompter.run]]
+	 */
+	public strategy: MessagePrompterBaseStrategy;
+
+	/**
+	 * Constructor for the [[MessagePrompter]] class
+	 * @param message The message to send.
+	 * @param strategy The strategy name or Instance to use
+	 * @param strategyOptions The options that are passed to the strategy
+	 */
+	public constructor(message: MessagePrompterMessage | MessagePrompterBaseStrategy, strategy?: string, strategyOptions?: Record<string, unknown>) {
+		let strategyToRun: MessagePrompterBaseStrategy | undefined = undefined;
+
+		if (message instanceof MessagePrompterBaseStrategy) {
+			strategyToRun = message as MessagePrompterBaseStrategy;
+		} else {
+			const mapStrategy = MessagePrompter.strategies.get(strategy ?? MessagePrompter.defaultStrategy);
+
+			if (!mapStrategy) {
+				throw new Error('No strategy provided');
+			}
+
+			strategyToRun = new mapStrategy(message, strategyOptions);
+		}
+
+		this.strategy = strategyToRun;
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter An author object to validate or a {@link https://discord.js.org/#/docs/main/stable/typedef/CollectorFilter CollectorFilter} predicate callback.
+	 * @param explicitReturn Wheter or not you wan't an explicit return object
+	 */
+	public run(channel: TextChannel | NewsChannel | DMChannel, authorOrFilter: User | CollectorFilter) {
+		return this.strategy.run(channel, authorOrFilter);
+	}
+
+	/**
+	 * The available strategies
+	 */
+	// @ts-expect-error 2322
+	public static strategies: Map<string, Constructor<MessagePrompterBaseStrategy>> = new Map([
+		['confirm', MessagePrompterConfirmStrategy],
+		['number', MessagePrompterNumberStrategy],
+		['reaction', MessagePrompterReactionStrategy],
+		['message', MessagePrompterMessageStrategy]
+	]);
+
+	/**
+	 * The default strategy to use
+	 */
+	public static defaultStrategy = 'confirm';
+
+	/**
+	 * The default strategy options
+	 */
+	public static defaultStrategyOptions: IMessagePrompterStrategyOptions = {
+		timeout: 10 * 1000,
+		explicitReturn: false
+	};
+}
+
+/**
+ * A type to extend multiple discord types and simplify usage in [[MessagePrompter]]
+ */
+export type MessagePrompterMessage = APIMessageContentResolvable | (MessageOptions & { split?: false }) | MessageAdditions;
+
+// strategy options
+export interface IMessagePrompterStrategyOptions {
+	timeout: number;
+	explicitReturn: boolean;
+}
+
+export interface IMessagePrompterConfirmStrategyOptions extends IMessagePrompterStrategyOptions {
+	confirmEmoji?: string | EmojiIdentifierResolvable;
+	cancelEmoji?: string | EmojiIdentifierResolvable;
+}
+
+export interface IMessagePrompterNumberStrategyOptions extends IMessagePrompterStrategyOptions {
+	start?: number;
+	end?: number;
```

Comment:
```suggestion
	minimum?: NumberRange;
	maximum?: NumberRange;
```

To match args syntax. Update TSDoc accordingly as well.

Reply:
I find minimum, maximum a wrong term as it generates a array of numbers between start & end

Reply:
Fair nuf